post-scan-actions/gcp-python-slack-notification/docs/deploy-tutorial.md

```diff
@@ -0,0 +1,32 @@
+# Cloud One File Storage Security Post Scan Action for GCP - Slack notifications
+
+## Overview
+
+<walkthrough-tutorial-duration duration="10"></walkthrough-tutorial-duration>
+
+This tutorial will guide you to deploy a Google Cloud function to push Slack notifications.
+
+## Project Setup
+
+Copy the execute the script below to set the project ID where the storage stack is deployed.
+
+<walkthrough-project-setup></walkthrough-project-setup>
+
+> If using a local machine and not Google Cloud Shell, use `gcloud auth application-default login` to login with gcloud CLI.
+
+```
+gcloud config set project <walkthrough-project-id/>
+```
+
+## Deploy Google Cloud function to push Slack notifications
+
+1. Setup the serverless.yml file with your `environment` variables, like `SLACK_URL`, `SLACK_CHANNEL`, `SLACK_USERNAME`
```

Comment:
Consider using [CLI parameters](https://www.serverless.com/framework/docs/guides/parameters#cli-parameters) to specify their values.

Reply:
Love the params suggestion ❤️. I am incorporating the **${param:XXX}** such that it takes CLI parameters and falls back to params.<stage> parameters, if CLI parameter is not found.

Reply:
I thought CLI parameters are enough for deployment. Is there any use case where the deployment cannot take CLI parameters and needs to use `params` in the template?

Reply:
You don't need to define a stage for reducing the hard-coded (or default?) values. Like the following example, `deploymentRegion` will be `us-east-1` if CLI parameters don't contain `DEPLOYMENT_REGION`.
```yaml
custom:
  eventType: providers/cloud.pubsub/eventTypes/topic.publish
  deploymentRegion: ${param:DEPLOYMENT_REGION, 'us-east1'}
```

What I want to reduce is the `<XXX_XXX>` in the serverless template. From the points of automation, replacing those strings in a file is much harder than passing them through CLI.

Reply:
Customer can still pass them as CLI arguments if they don't want to hardcode it inside the serverless.yml file. `params` enables both CLI and stage variables and the customer can choose what variables they need to set in stage so when they don't pass a CLI argument, it would fail over to stage params.

Reply:
This has a precondition that they already hardcoded all `<XXX>` in `serverless.yml`. If they didn't do it, the "fail over" doesn't work. Those variables are inputs for deploying the resources, and serverless framework can help customers deploy the resources with the same codes but different inputs. That's one of the advantages of using IaC tool, and I really like the choice. Given that the tool already provides the capability, why would we still keep "hardcode" way for specifying inputs? Is there any scenario or advantage that I don't see?

Reply:
Yup, so I'd like to know under which scenario users will choose to hardcode values in the template. Another thing is the stage for serverless framework doesn't seem to be designed for "fail over". If we don't have other "stage", I don't think we need that. If there's no such use case, I prefer we remove the stage and keep only CLI parameters to make it simpler.

Reply:
That might be helpful for teams sharing the same project but different channels. If you insist to hardcode default values, another choice is to use [dotenv](https://www.serverless.com/framework/docs/environment-variables) with `.env` file. Then users don't need to modify `serverless.yml` for every environment, which I don't think is a good practice.

Stage parameter is designed for deploying the templates to different stages with their own parameters. It indeeds have fail over behavior, but it doesn't mean we should use it just because of fail over.

Reply:
I am not sure if it would be a good idea to add new dependencies, like dotenv,  to the code to implement a feature that is available in the vanilla serverless framework. Users can simply use the CLI params to pass/override to the script runtime.

Thank you for your time and feedback. It is much appreciated.